pairinteraction_gui/pairinteraction/app.py

```diff
@@ -3025,13 +3049,28 @@ def startCalc(self):
                 # undefined one thread per CPU is used.
                 ompthreads = {} if self.numprocessors == 0 else {"OMP_NUM_THREADS": str(self.numprocessors)}
 
-                self.proc = subprocess.Popen(
-                    [path_cpp, "-c", self.path_config, "-o", self.path_cache],
-                    stdout=subprocess.PIPE,
-                    stderr=subprocess.STDOUT,
-                    cwd=self.path_workingdir,
-                    env=dict(os.environ, **ompthreads),
-                )
+                if self.ui.checkbox_use_python_api.isChecked():
+                    self.proc = subprocess.Popen(
```

Comment:
I thought the whole point of migrating to the Python interface was that we don't have to fork out a separate process anymore.

Reply:
I agree that avoiding forking out a separate process would be nice. However, this does not need to be an immediate goal. For me, the main motivation for getting rid of the old code is increasing maintainability and making it possible to integrate features in the GUI that are currently only available in the Python interface (e.g., supporting strontium). Johanne's approach has the benefit of being a drop-in replacement for the old backend that does not require lots of changes to the existing code. Thus, the risk of breaking existing functionality is reduced. In addition, as it turned out that migrating to the new Python interface is a rather convoluted task, I like the idea of splitting it into several smaller pull requests (e.g. first incorporating Johanne's changes, second avoiding forking out a separate process and maybe rewriting some code in C++, third adding strontium to the GUI, etc.) - especially as this would allow us to get direct user feedback on the current development status.

Reply:
we are now using a qthread to run the simulations with the new python backend